### DISCUSSION TITLE
Should I upgrade to SSM-D

### DISCUSSION THREAD

**Discussion by BasKloet** [ghoti57/evofw3#28] [Q&A]
I'm looking for advice on improving the functionality and reliability of my current set-up, before I decide to spend money on the SSM-D and then finding out that my current issues are not actually based around (that part of) my hardware. 

First of all, my current Honeywell setup:
I have a Honeywell Evohome Wireless thermostat, an RFG100 and a wireless receiver (I believe the R8810A118) and I use a NanoCUL for evofw3.

I had everything working to a point where I could use the NanoCUL to get all relevant heating data including zone information like the current thermostat setpoint, but I was unable send commands to change any (thermostat) settings. Therefor I used the Honeywell API to control the thermostat in my home automation. Recently however something changed, and I don't know what, which led to the zone information no longer being correctly received (current/setpoint are both consistently 0 degrees C). I still get updates from the main heating system, like DHW temperature, so it's not completely broken. I've tried updating the firmware and tuning the receiver, but that made no difference. I also see multiple messages from evohome about corrupt packets in my domoticz log. After reading about the limitations of the NanoCUL, I'm considering upgrading to the SSM-D, but I would like to know if that has a reasonable change to fix my issues before spending the money on the device + shipping/VAT to the Netherlands.

So my questions are:
1. Given the described situation, would upgrading to the SSM-D be a good idea or is there something I can still do with my current hardware?
2. If the outcome is that I should upgrade to the SSM-D, then @ghoti57: can you give me an estimate of when you will be able to ship them again and a ballpark figure of the total amount I can expect to pay for the device + shipping + VAT?

Note: I'm not a big fan of the Honeywell API, both from a privacy perspective and because the API frequently gives (authentication) errors for no apparent reason. So in an ideal world I would like to create a solution where I don't need the Honeywell API for anything anymore and can preferably even remove the RFG100 from my setup.

Thanks,
Bas

**Comment by ghoti57** [comment]
I don't have enough information to say whether you should upgrade.

There are a number of questions.
1. Which 'nanocul' are you using?  Where/when did you acquire it?
2. What version of evofw3 are you using?
3. How did you load evofw3?
4. Did you have any problems when you first started to use evofw3?
5. Do you have any idea at all about what changed?
6. Are you committed to using Domoticz for your home automation?  There is significant activity for a custom plugin in Home Assistant that totally bypasses the Honeywell GUI. (See here for info 
https://github.com/zxdavb/evohome_cc/wiki)

The thing that concerns me most about your setup is your apparent inability to get any TX working.  While evofw3 uses different RX methods on Nanocul and SSM-D, they share the same method for TX.

Have you tried any tests outside Domoticz with the Nanocul connected to a serial Comms program such as putty?


I should be shipping again early May.  The postage costs are as I post on the shop.  VAT is at the standard rate in NL.  I don't include the value of UK VAT (20%) on the customs label to minimise your charges.  The postage is also technically subject to NL VAT but it appears that it depends which individual processes the package as to whether that gets applied.    PostNL also have a handling charge which you should be able to find on their website.  I believe this charge is different depending on whether you pay online or on delivery.

**Reply by BasKloet** [reply]
First of all, thank you for your extensive reply, I will try to answer all your questions as well as I can.

1. I got the [nanoCUL USB Stick FTDI CC1101 868MHz](https://www.amazon.de/dp/B07T3H8VF2/ref=pe_27091401_487027711_TE_SCE_3p_dp_1) from Amazon (see link)
2. I'm using version 0.7.1 of evofw3 since about a week, before that I was using 0.7.0. 
3. I loaded the firmware using the Arduino IDE  on Manjaro Linux and run the stick itself from a raspberry PI
4. I've been using the stick for about 5 months now and I've never been able to set the system mode through the stick, but I was able to receive zone information until about 2 weeks ago. For most of the time I was using evoGateway to communicate with the stick using mqtt, and I even filed an [issue](https://github.com/smar000/evoGateway/issues/40) there because I couldn't sent commands.
5. I have no idea unfortunately, though as described in point 4 there were some issues from the start. When I noticed that I was no longer receiving zone information, I first upgraded the firmware to 0.7.1, but that didn't help. I then replaced the USB hub that was between my PI and the stick, to see if that was the culprit. I finally tried tuning the frequency, but that did not help either.
6. If I had to set everything up from scratch again, I would use Home Assistant, but right now I have such a good working system with Domoticz that I don't want to switch my current setup. I was using [evoGateway](https://github.com/smar000/evoGateway), so that was not dependent on Domoticz and I only switched to the Domoticz plugin to see if that would fix my problem.

_**Note:**_ While writing the above reply, I decided to try evoGateway once more and I now see Zone information in mqtt again, so maybe all my changes did have an effect, just not in the Domoticz plugin. I will do some more investigation and let you know the results.

**Reply by BasKloet** [reply]
Update, it seems like it was indeed some combination of the stick location, the new usb hub, the tuning and the latest firmware, but it works! Not in the Domoticz plugin, but I can easily live without that. I just tested changing the system mode over mqtt and it works, which is something that I had already given up on!

So in short, everything works as I want it again, so I will not need the SSM-D for now. Thank you for the quick and complete reply and for creating this great firmware.
